src/main/java/orca/flukes/MouseMenus.java

```diff
@@ -520,6 +521,57 @@ public void setPoint(Point2D point) {
 
 	}
 	
+	public static class NodeInsertSSHKeyItem extends JMenuItem implements NodeMenuListener<OrcaNode, OrcaLink>, MenuPointListener {
+	    OrcaResource node;
+	    VisualizationViewer<OrcaNode, OrcaLink> visComp;
+	    Point2D point;
+
+	    public NodeInsertSSHKeyItem(final JFrame frame) {
+	        super("Insert guest SSH key ...");
+	        this.addActionListener(new ActionListener() {
+	            public void actionPerformed(ActionEvent e) {
+	                try {
+	                    // display key dialog

```

Comment:
This looks like exactly the same code as above (in GUIUnifiedState.java) ?

Reply:
It is similar - in mousemenus this deals with a single node and above it deals with all nodes in a slice.